### ISSUE TITLE
[Bug] `Lint/UselessAssignment` false positive with `in` operator

### ISSUE THREAD

**Issue by jtannas** [rubocop/rubocop#13984]
`Lint/UselessAssignment` throws a false positive when a variable is being used with the `in` operator.

--------

## Expected behavior

I want rubocop to correctly recognize that a variable is being used.

## Actual behavior

It flags it as an error,
 
```
$ bundle exec rubocop --debug --only Lint/UselessAssignment test.rb
For /home/jtannas/rubocop: configuration from /home/jtannas/rubocop/.rubocop.yml
Default configuration from /home/jtannas/.asdf/installs/ruby/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.74.0/config/default.yml
Plugin configuration from /home/jtannas/.asdf/installs/ruby/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.74.0/config/internal_affairs.yml
Plugin configuration from /home/jtannas/.asdf/installs/ruby/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-performance-1.24.0/config/default.yml
Plugin configuration from /home/jtannas/.asdf/installs/ruby/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-rake-0.7.1/config/default.yml
Plugin configuration from /home/jtannas/.asdf/installs/ruby/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-rspec-3.5.0/config/default.yml
Inheriting configuration from /home/jtannas/rubocop/.rubocop_todo.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /home/jtannas/rubocop/test.rb
Loading cache from /home/jtannas/.cache/rubocop_cache/70312f5d91ad4ddae87e78d78bb5fbef356a4d06/f7a68b00a409cc5d84b9a3c28a9b0fbd5cdfd4b3/26d33734f42f9c14bfa8fcf6ce745b30aaa97cc4
W

Offenses:

test.rb:3:1: W: [Correctable] Lint/UselessAssignment: Useless assignment to variable - pattern.
pattern = { a: 1 }
^^^^^^^

1 file inspected, 1 offense detected, 1 offense autocorrectable
Finished in 0.32578456600003847 seconds
```

## Steps to reproduce the problem

```ruby
pattern = { a: 1 }
is_match = { a: 1 } in pattern
```

## RuboCop version
```
$ rubocop -V
1.74.0 (using Parser 3.3.7.1, rubocop-ast 1.38.1, analyzing as Ruby 3.2, running on ruby 3.4.2) [x86_64-linux]
  - rubocop-performance 1.24.0
  - rubocop-rake 0.7.1
  - rubocop-rspec 3.5.0
```

**Comment by jtannas** [comment]
My bad - I misunderstood how `in` works and Rubocop is correct here.
https://bugs.ruby-lang.org/issues/21184
```ruby
irb(main):001> pattern = { a: 1 }
=> {a: 1}
irb(main):002> 3 in pattern
=> true
irb(main):003> pattern
=> 3
```
